sushichef.py

```diff
@@ -45,35 +49,84 @@ class MyChef(SushiChef):
+                |--- |--- Real Pasifik 2 introducing Chef Alexis Tahiapuhe of Tahiti (source_id = file-real pasifik 2 introducing chef lela bolobolo of fiji.mp4)
+                |--- Pacific Islands Arts and Culture(source_id = dir_pacific_islands_arts_and_culture)
+                |--- |--- Cook Islands National Cultural Policy 10 July 2017_final english (File)
+                |--- Teaching Resources and Classroom Activities
+                Environment (source_id = dir-environment)
+                |--- Adapting to Climate Change
+                |--- |--- Action Against Climate Change Tuvalu Water and climate change
+                |--- Climate Change Info                
+                |--- |--- Animated Pacific Island Climate Change Videos
+                ...
+            Returns: ChannelNode
         """
-        Creates ChannelNode and build topic tree
-        Args:
-          - args: arguments passed in during upload_channel (currently None)
-          - kwargs: extra argumens and options not handled by `uploadchannel`.
-            For example, add the command line option   lang="fr"  and the string
-            "fr" will be passed along to `construct_channel` as kwargs['lang'].
-        Returns: ChannelNode
-        """
-        channel = self.get_channel(*args, **kwargs)  # Create ChannelNode from data in self.channel_info
+        LOGGER.info("Constructing channel from {}...".format(BASE_URL))
+        channel = self.get_channel(*args, **kwargs)                         # Creates ChannelNode from data in self.channel_info
+        LOGGER.info('   Writing {} Folder...'.format(CHANNEL_NAME))    
+        endpoint = BASE_URL + "content/"
+        scrape_content(endpoint, channel)
+        raise_for_invalid_channel(channel)                                  # Check for errors in channel construction
+        return channel
 
-        # TODO: Replace next line with chef code
-        raise NotImplementedError("constuct_channel method not implemented yet...")
+""" Helper Methods """
+###########################################################
 
-        raise_for_invalid_channel(channel)  # Check for errors in channel construction
+def read_source(url):
+    html = downloader.read(url)    
+    return BeautifulSoup(html, "html.parser")    
 
-        return channel
+def replace_all(text, replacements):
+    for condition, replacement in replacements.items():
+        text = text.replace(condition, replacement)
+    return text
+
+def scrape_content(endpoint, channel, existingNode=None):
+    replacements = {" ": "%20", "#": "%23"} 
+    content = read_source(endpoint)
+    attributes = content.find("tbody").find_all("td", "text-xs-left")
+
+    for attribute in attributes:
+        source_id = attribute.attrs["data-sort-value"]
+
+        # Check if it is mp4 file
+        if source_id.endswith(".mp4"):
+            video_info = attribute.find("a")
+            video_title = str(video_info.string)
```

Comment:
Could you change this to `video_title, _ext = splitext(str(video_info.string))` to remove the .mp4 ending?